From: Eric B Munson <emunson@mgebm.net>
Subject: hugetlb: add generic definition of NUMA_NO_NODE
References: fate#311647 bnc#691979
Patch-Mainline: yes

Signed-off-by: Thomas Renninger <trenn@suse.de>

This patch backports commit 4e25b2576efda24c02e2d6b9bcb5965a3f865f33
by Lee Schermerhorn <lee.schermerhorn@hp.com>

Signed-off-by: Eric B Munson <ebmunson@us.ibm.com>

Original commit message:

    Move definition of NUMA_NO_NODE from ia64 and x86_64 arch specific headers
    to generic header 'linux/numa.h' for use in generic code.  NUMA_NO_NODE
    replaces bare '-1' where it's used in this series to indicate "no node id
    specified".  Ultimately, it can be used to replace the -1 elsewhere where
    it is used similarly.
---
 arch/ia64/include/asm/numa.h    |    2 --
 arch/x86/include/asm/topology.h |    7 +++++++
 include/linux/numa.h            |    2 ++
 3 files changed, 9 insertions(+), 2 deletions(-)

--- a/arch/ia64/include/asm/numa.h
+++ b/arch/ia64/include/asm/numa.h
@@ -22,8 +22,6 @@
 
 #include <asm/mmzone.h>
 
-#define NUMA_NO_NODE	-1
-
 extern u16 cpu_to_node_map[NR_CPUS] __cacheline_aligned;
 extern cpumask_t node_to_cpu_mask[MAX_NUMNODES] __cacheline_aligned;
 extern pg_data_t *pgdat_list[MAX_NUMNODES];
--- a/arch/x86/include/asm/topology.h
+++ b/arch/x86/include/asm/topology.h
@@ -35,6 +35,13 @@
 # endif
 #endif
 
+/*
+ * to preserve the visibility of NUMA_NO_NODE definition,
+ * moved to there from here.  May be used independent of
+ * CONFIG_NUMA.
+ */
+#include <linux/numa.h>
+
 #ifdef CONFIG_NUMA
 #include <linux/cpumask.h>
 #include <asm/mpspec.h>
--- a/include/linux/numa.h
+++ b/include/linux/numa.h
@@ -12,4 +12,6 @@
 
 #define NUMA_NO_NODE	(-1)
 
+#define	NUMA_NO_NODE	(-1)
+
 #endif /* _LINUX_NUMA_H */
